packages/twenty-server/src/engine/workspace-manager/workspace-sync-metadata/utils/should-exclude-from-workspace-api.util.ts

```diff
@@ -0,0 +1,28 @@
+import { BaseWorkspaceEntity } from 'src/engine/twenty-orm/base.workspace-entity';
+import { metadataArgsStorage } from 'src/engine/twenty-orm/storage/metadata-args.storage';
+
+import { isGatedAndNotEnabled } from './is-gate-and-not-enabled.util';
+
+export const shouldExcludeFromWorkspaceApi = (
+  objectMetadataItem: { standardId?: string | null | undefined },
+  standardObjectMetadataDefinitions: (typeof BaseWorkspaceEntity)[],
+  workspaceFeatureFlagsMap: Record<string, boolean>,
+): boolean => {
+  const workspaceEntity = standardObjectMetadataDefinitions.find((entity) => {
```

Comment:
nit: suggestion to avoid calling filterEntities twice
```suggestion
    const entityMetadata = standardObjectMetadataDefinitions
    .map((entity) => metadataArgsStorage.filterEntities(entity))
    .find((meta) => meta?.standardId === objectMetadataItem.standardId);

  if (!entityMetadata) {
    return false; // Don't exclude non-workspace entities
  }
```